packages/openapi-ts/test/__snapshots__/test/generated/v3_angular_transform/types.gen.ts.snap

```diff
@@ -0,0 +1,132 @@
+// This file is auto-generated by @hey-api/openapi-ts
+
+/**
+ * This is a model that contains a some dates
+ */
+export type SimpleModel = {
+    id: number;
+    name: string;
+    readonly enabled: boolean;
+};
+
+/**
+ * This is a model that contains a some dates
+ */
+export type ModelWithDates = {
+    id: number;
+    name: string;
+    readonly enabled: boolean;
+    readonly modified: Date;
+    readonly expires?: Date;
+};
+
+export function ModelWithDates(data: any): ModelWithDates {
+    if (data?.modified) {
+        data.modified = new Date(data.modified);
+    }
+    if (data?.expires) {
+        data.expires = new Date(data.expires);
+    }
+    return data;
+}
+
+/**
+ * This is a model that contains a some dates and arrays
+ */
+export type ParentModelWithDates = {
+    id: number;
+    readonly modified?: Date;
+    items?: Array<ModelWithDates>;
+    item?: ModelWithDates;
+    simpleItems?: Array<SimpleModel>;
+    simpleItem?: SimpleModel;
+    dates?: Array<(Date)>;
+    strings?: Array<(string)>;
+};
+
+export function ParentModelWithDates(data: any): ParentModelWithDates {
```

Comment:
Would be nice to add proper types here

Reply:
There's not much point, it's an internal function called by a part of the code that doesn't know the type because it's a response body, so you wouldn't get any compiler benefits. The mutations also benefit from `any` because we don't have to do any casting during the transform, though that might not be an issue

I'm treating these similar to validation/parser logic in my mind. For instance Zod.parse is always `unknown->MyType` albeit I'll cede that it checks every key in that situation

Reply:
I was thinking it could help with catching incorrect transforms in tests... why not use `unknown` over `any` btw?

Reply:
You could be right, there's no harm in declaring the type on the input and I can't think of a reason it should cause compile issues unless there's a real issue in the transform. I was mostly just going after the validator/parser pattern that tools like Zod use, but it is slightly different here

There is a good reason to use `any` over `unknown` though, which is unknown means you can't chain up properties without do a `"x" in y` check or using a type assertion first so the codegen would be way more verbose and difficult to generate. Any is like disabling typescript

I'm happy to switch things around to take the expected type as input. The main impact would be on the request files but I suspect since their type definition is `unknown -> unknown`  that will all work fine

Reply:
OH actually, I remember why I did it this way - static typed input would be nice, but I didn't want _any_ risk of an unexpected response causing a `property x of undefined` type error, and so every single access chain uses the optional chaining operator. If we gave it a static input type then is there a chance prettier/eslint could strip the "unnecessary optional chaining" since the input type is known and properties in each chain aren't technically optional?